auth-server/src/controllers/public.ts

```diff
@@ -0,0 +1,70 @@
+import express, { Request, Response } from 'express';
+import { DecodedToken } from '../middleware/requireLogin';
+const hri = require('human-readable-ids').hri;
+const publicRouter = express.Router();
+
+interface PublicBeacon {
+  beaconId: string;
+  nickname: string;
+}
+
+let publicBeacons: PublicBeacon[] = [];
+
+const generateNickname = () => {
+  return hri.random();
+};
+
+publicRouter.get('/', (request: Request, response: Response) => {
+  response.status(200).send(publicBeacons);
+});
+
+publicRouter.post(
+  '/',
+  (request: Request & DecodedToken, response: Response) => {
```

Comment:
Please use async/await syntax here. Supertest silently ignores exceptions if async/await is not used, which makes debugging hard.